projects/storefrontlib/src/cms-components/product/stock-notification/stock-notification.component.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  Component,
+  OnInit,
+  OnDestroy,
+  ChangeDetectionStrategy,
+} from '@angular/core';
+import {
+  UserInterestsService,
+  UserNotificationPreferenceService,
+  AuthService,
+  OCC_USER_ID_ANONYMOUS,
+  NotificationPreference,
+  NotificationType,
+  Product,
+  GlobalMessageService,
+  TranslationService,
+  GlobalMessageType,
+} from '@spartacus/core';
+import { Observable, Subscription, combineLatest } from 'rxjs';
+import { map, filter, tap, first } from 'rxjs/operators';
+import { CurrentProductService } from '../current-product.service';
+import { ModalService } from '../../../shared/components/modal/modal.service';
+import { StockNotificationDialogComponent } from './stock-notification-dialog/stock-notification-dialog.component';
+
+@Component({
+  selector: 'cx-stock-notification',
+  templateUrl: './stock-notification.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class StockNotificationComponent implements OnInit, OnDestroy {
+  subscribed$: Observable<boolean>;
+  prefsEnabled$: Observable<boolean>;
+  outOfStock$: Observable<boolean>;
+  subscribeSuccess$: Observable<boolean>;
+  unsubscribeLoading$: Observable<boolean>;
```

Comment:
This name is not very obvious, what about `isRemoveProdutInterestLoading$` or something like that?